packages/server/controllers/auth/login.js

```diff
@@ -5,7 +5,15 @@ const { createToken } = require("../../services/token.service");
 const { validatePassword } = require("../../utils/password");
 const logger = require("../../utils/logger");
 const error = require("../../errorResponse.json");
-
+// Load blacklisted domains from environment variable
+const blacklistedDomains = process.env.BLACKLISTED_DOMAINS
+  ? process.env.BLACKLISTED_DOMAINS.split(',').map(domain => domain.trim().toLowerCase())
+  : [];
+  // Check if domain is blacklisted
+const isDomainBlacklisted = (email) => {
+  const domain = email.split("@")[1]?.toLowerCase();
+  return blacklistedDomains.includes(domain);
+};
```

Comment:
Wouldn't it be better to create a common function in `packages/server/services/auth` directory and use that in `signup.js and `login.js`?